download.py

```diff
@@ -22,30 +26,87 @@
 if force_automodel:
     print(f"Using AutoModel for {model_name} to instantiate model")
 
-print(f"Downloading model {model_name} from huggingface model hub")
-config = AutoConfig.from_pretrained(model_name)
-model_type = config.to_dict()['model_type']
+onnx_runtime = os.getenv('ONNX_RUNTIME')
+if not onnx_runtime:
+    onnx_runtime = "false"
 
-if model_type is not None and model_type == "t5":
-    SentenceTransformer(model_name, cache_folder=model_dir)
-    with open(f"{model_dir}/model_name", "w") as f:
-        f.write(model_name.replace("/", "_"))
-else:
-    if config.architectures and not force_automodel:
-        print(f"Using class {config.architectures[0]} to load model weights")
-        mod = __import__('transformers', fromlist=[config.architectures[0]])
-        try:
-            klass_architecture = getattr(mod, config.architectures[0])
-            model = klass_architecture.from_pretrained(model_name)
-        except AttributeError:
-            print(f"{config.architectures[0]} not found in transformers, fallback to AutoModel")
-            model = AutoModel.from_pretrained(model_name)
-    else:
-        model = AutoModel.from_pretrained(model_name)
+onnx_cpu_arch = os.getenv('ONNX_CPU')
+if not onnx_cpu_arch:
+    onnx_cpu_arch = "arm64"
+
+print(f"Downloading MODEL_NAME={model_name} with FORCE_AUTOMODEL={force_automodel} ONNX_RUNTIME={onnx_runtime} ONNX_CPU={onnx_cpu_arch}")
+
+def download_onnx_model(model_name: str, model_dir: str):
+    # Download model and tokenizer
+    onnx_path = Path(model_dir)
+    ort_model = ORTModelForFeatureExtraction.from_pretrained(model_name, from_transformers=True)
+    # Save model
+    ort_model.save_pretrained(onnx_path)
+
+    def save_to_file(filepath: str, content: str):
+        with open(filepath, "w") as f:
+            f.write(content)
+
+    def save_quantization_info(arch: str):
+        save_to_file(f"{model_dir}/onnx_quantization_info", arch)
+
+    def quantization_config(onnx_cpu_arch: str):
+        if onnx_cpu_arch.lower() == "avx512_vnni":
+            print("Quantize Model for x86_64 (amd64) (avx512_vnni)")
+            save_quantization_info("AVX-512")
+            return AutoQuantizationConfig.avx512_vnni(is_static=False, per_channel=False)
+        if onnx_cpu_arch.lower() == "arm64":
+            print(f"Quantize Model for ARM64")
+            save_quantization_info("ARM64")
+            return AutoQuantizationConfig.arm64(is_static=False, per_channel=False)
+        # default is AMD64 (AVX2)
+        print(f"Quantize Model for x86_64 (amd64) (AVX2)")
+        save_quantization_info("amd64 (AVX2)")
+        return AutoQuantizationConfig.avx2(is_static=False, per_channel=False)
```

Comment:
Quantizing dynamically is a good for now but I think it would be nice to be able to do it statically, perhaps as an option so that we have many duplicated Docker images: one dynamic and one static for each model